deployment/cfn/application.py

```diff
@@ -270,29 +299,75 @@ def create_load_balancer(self, app_server_lb_security_group):
 
     def create_auto_scaling_resources(self, app_server_security_group,
                                       app_server_lb):
-        app_server_launch_config_name = 'lcAppServer'
+        self.add_condition('BlueCondition', Equals('Blue', Ref(self.color)))
+        self.add_condition('GreenCondition', Equals('Green', Ref(self.color)))
 
-        app_server_launch_config = self.add_resource(
+        blue_app_server_launch_config = self.add_resource(
             asg.LaunchConfiguration(
-                app_server_launch_config_name,
+                'lcAppServerBlue',
+                Condition='BlueCondition',
                 ImageId=Ref(self.app_server_ami),
                 IamInstanceProfile=Ref(self.app_server_instance_profile),
                 InstanceType=Ref(self.app_server_instance_type),
                 KeyName=Ref(self.keyname),
-                SecurityGroups=[Ref(app_server_security_group)]
+                SecurityGroups=[Ref(app_server_security_group)],
+                UserData=Base64(
```

Comment:
Thanks, that's pretty cool. Since cloud-init only runs the first time you launch an instance, does that mean we need to relaunch a new app instance if there was a tiler change?